vllm/model_executor/weight_utils.py

```diff
@@ -97,32 +98,52 @@ def prepare_hf_model_weights(
     else:
         hf_folder = model_name_or_path
     hf_weights_files = glob.glob(os.path.join(hf_folder, allow_patterns))
-    if not use_safetensor:
+    if not use_safetensors:
         hf_weights_files = [
             x for x in hf_weights_files if not x.endswith("training_args.bin")
         ]
 
-    if len(hf_weights_files) == 0 and use_safetensor:
-        logger.warning("No *.safetensors files found, "
-                       "fall back to *.bin files")
+    if len(hf_weights_files) == 0 and use_safetensors and fall_back_to_pt:
```

Comment:
Can we raise an error when no weight file matching the load format is found? This would be much better than silently uninitializing the model weights.